### PR TITLE
fix(imports): #MC-138 fix hour change when importing ical

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
fix 2h offset when importing ical
import error will appear if ther is a problem on the time :
image

## Checklist tests
import ical and check events times.
Check if it works with and without the environment variable TZ: "Europe/Paris"

## Issue ticket number and link
#MC-138 https://entsupport.gdapublic.fr/browse/MC-138

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by aymericdevriesere — APPROVED** [review]

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 1]
```diff
@@ -218,8 +221,17 @@ private void setEventDates(VEvent event, JsonObject jsonEvent) {
         boolean allDay = dtStartValue.equals("DATE");
         Date startDate = event.getStartDate().getDate();
         Date endDate = event.getEndDate().getDate();
-        String startMoment = MOMENT_FORMAT.format(startDate);
-        String endMoment = MOMENT_FORMAT.format(endDate);
+
+        java.util.Date newStartDate = null;
```
Comment (sjollois) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:225]: c'est quoi la diff entre Date et java.util.Date ?

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 2]
```diff
@@ -218,8 +221,17 @@ private void setEventDates(VEvent event, JsonObject jsonEvent) {
...
+        java.util.Date newEndDate = null;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_DATE_FORMAT).parse(startDate.toString());
+            newEndDate = new SimpleDateFormat(DateUtils.ICAL_DATE_FORMAT).parse(endDate.toString());
+        } catch (ParseException e) {
+            e.printStackTrace();
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:231]: utilise nos conventions de log plutôt ce qu'on te propose habituellement (e.g printStacktrace)

Et que fais-tu si ça échoue ?

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by sjollois — APPROVED** [review]
Petit commentaire d'incompréhension mais ok sinon

**Review by BastienBenard — APPROVED** [review]

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java (hunk 1 above)
Comment (alicedraillard) [reply src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:225]: C'est pour éviter d'utiliser net.fortuna.ical4j.model.Date qui est utilisé dans le reste du module

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 3]
```diff
@@ -211,15 +218,28 @@ private String encodeString(String toEncode) {
...
+        java.util.Date newEndDate = null;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_DATE_FORMAT).parse(startDate.toString());
+            newEndDate = new SimpleDateFormat(DateUtils.ICAL_DATE_FORMAT).parse(endDate.toString());
+        } catch (ParseException e) {
+            String message = String.format("[Calendar@%s::jsonEventsToIcsContent] An error has occured" +
+                            " during ical import: %s",
+                    this.getClass().getSimpleName(), e.getMessage());
+            log.error(message, e.getMessage());
+            throw new UnhandledEventException("Problème de gestion des horaires de l'évènement.");
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:239]: internationaliser avec une clé "slot.event.problem"

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 4]
```diff
@@ -211,15 +218,28 @@ private String encodeString(String toEncode) {
...
         boolean allDay = dtStartValue.equals("DATE");
         Date startDate = event.getStartDate().getDate();
         Date endDate = event.getEndDate().getDate();
-        String startMoment = MOMENT_FORMAT.format(startDate);
-        String endMoment = MOMENT_FORMAT.format(endDate);
+
+        java.util.Date newStartDate = null;
+        java.util.Date newEndDate = null;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_DATE_FORMAT).parse(startDate.toString());
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:232]: Dans ta checklist de test, prendre en compte l'env de docker

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**